### ISSUE TITLE
Only filter action params in view

### ISSUE THREAD

**Issue by korczis** [zalmoxisus/redux-devtools-filter-actions#7]
It will be nice if it will be possible to only filter action parameters.

I have for example action FILE_DOWNLOAD which is used for downloading data from backend. I have to use this middleware to prevent freeze of browser.

For me it will be sufficient see something like `data: <filtered>` and thus do not filter entire action from devtools. Is this possible?

PS: This can be useful for filtering out sensitive params too.

**_Example**_

```
const blacklist = [
  {
   name: 'FILE_DOWNLOAD_SUCCESS', 
   params: ['data']
  }
]

<DockMonitor toggleVisibilityKey="ctrl-H" changePositionKey="ctrl-Q">
  <FilterMonitor blacklist={blacklist}>
    <LogMonitor />
  </FilterMonitor>
</DockMonitor>
```

**Comment by korczis** [comment]
@zalmoxisus: It will be even better if one could pass function which will return if the event should be filtered or some parts of it.

**Comment by zalmoxisus** [comment]
Hey @korczis,

That would be useful indeed. So we want to use it like that:

``` jsx
const filter = action => (
  action.type === 'FILE_DOWNLOAD_SUCCESS' ?
    Object.keys(action).reduce(function(previous, current) {
      if (current !== 'data') previous[current] = action[current];
      return previous;
    }, {})
  : action
);

<DockMonitor toggleVisibilityKey="ctrl-H" changePositionKey="ctrl-Q">
  <FilterMonitor filter={filter}>
    <LogMonitor />
  </FilterMonitor>
</DockMonitor>
```

**Comment by korczis** [comment]
Hey @zalmoxisus,
This looks good to me. It gives almost endless possibilities. 

IMHO it will be nice to have shorthand which allows pass names of actions and names of fields as in my example. Some people are scared to write functions ;-) Not me though! And code will be shorter and cleaner too. This is not required as it can be built on top of "filtering" function but it will be nice to have.

Anyway, any support of filtering function will be amazing.

**Comment by korczis** [comment]
@zalmoxisus Any updates?

**Comment by zalmoxisus** [comment]
@korczis, thanks for the heads up! I want to implement this feature here (and for Remote Redux DevTools  as well), but I'll be able to look into it just next week. If you can send a PR, it would be welcome.

**Comment by zalmoxisus** [comment]
Just released `v1.2.0`, so it can be used as following:

``` jsx
const actionsFilter = action => (
  action.type === 'FILE_DOWNLOAD_SUCCESS' && action.data ?
  { ...action, data: 'LONG_BLOB' } : action
);

<DockMonitor toggleVisibilityKey="ctrl-H" changePositionKey="ctrl-Q">
  <FilterMonitor actionsFilter={actionsFilter}>
    <LogMonitor />
  </FilterMonitor>
</DockMonitor>
```

Also you can filter states as well. More details in [the README](https://github.com/zalmoxisus/redux-devtools-filter-actions#containersdevtoolsjs). Please give it a try and let me know whether it fixes the issue.

**Comment by korczis** [comment]
@zalmoxisus Works like a charm, thank you for this!

PS: Sorry for late reply
